extra_data/reader.py

```diff
@@ -593,17 +593,33 @@ def union(self, *others):
         files = set(self.files)
         train_ids = set(self.train_ids)
 
+        run_num = self.run_metadata()[
+            "runNumber"] if self.is_single_run else None
+
+        proposal_num = self.run_metadata()[
+            "proposalNumber"] if self.is_single_run else None
```

Comment:
Older runs (the file format we call 0.5, before it actually had a version number) don't have this information stored in files. This is making the tests fail.

I guess the simplest thing is just to catch that so that `union` on old data always behaves like they were two separate runs, and the new feature is only for newer data. It's a bit unfortunate to have mysteriously different behaviour with different data, but we can probably live with it.

Reply:
added a check for these two keys for the current run format